tools/RAiDER/models/gmao.py

```diff
@@ -59,42 +62,60 @@ def _fetch(self, lats, lons, time, out, Nextra=2):
-
-        q = ds['qv'].array[
-            time_ind,
-            ml_min:(ml_max + 1),
-            lat_min_ind:(lat_max_ind + 1),
-            lon_min_ind:(lon_max_ind + 1)
-        ][0]
-        p = ds['pl'].array[
-            time_ind,
-            ml_min:(ml_max + 1),
-            lat_min_ind:(lat_max_ind + 1),
-            lon_min_ind:(lon_max_ind + 1)
-        ][0]
-        t = ds['t'].array[
-            time_ind,
-            ml_min:(ml_max + 1),
-            lat_min_ind:(lat_max_ind + 1),
-            lon_min_ind:(lon_max_ind + 1)
-        ][0]
-        h = ds['h'].array[
-            time_ind,
-            ml_min:(ml_max + 1),
-            lat_min_ind:(lat_max_ind + 1),
-            lon_min_ind:(lon_max_ind + 1)
-        ][0]
+        if time >= T0:
+            # open the dataset and pull the data
+            url = 'https://opendap.nccs.nasa.gov/dods/GEOS-5/fp/0.25_deg/assim/inst3_3d_asm_Nv'
+            session = pydap.cas.urs.setup_session('username', 'password', check_url=url)
+            ds = pydap.client.open_url(url, session=session)
+            q  = ds['qv'].array[
+                time_ind,
+                ml_min:(ml_max + 1),
+                lat_min_ind:(lat_max_ind + 1),
+                lon_min_ind:(lon_max_ind + 1)
+            ][0]
+
+            p = ds['pl'].array[
+                time_ind,
+                ml_min:(ml_max + 1),
+                lat_min_ind:(lat_max_ind + 1),
+                lon_min_ind:(lon_max_ind + 1)
+            ][0]
+            t = ds['t'].array[
+                time_ind,
+                ml_min:(ml_max + 1),
+                lat_min_ind:(lat_max_ind + 1),
+                lon_min_ind:(lon_max_ind + 1)
+            ][0]
+            h = ds['h'].array[
+                time_ind,
+                ml_min:(ml_max + 1),
+                lat_min_ind:(lat_max_ind + 1),
+                lon_min_ind:(lon_max_ind + 1)
+            ][0]
+
+        else:
+            root = 'https://portal.nccs.nasa.gov/datashare/gmao/geos-fp/das/Y{}/M{:02d}/D{:02d}'
+            base = f'GEOS.fp.asm.inst3_3d_asm_Nv.{time.strftime("%Y%m%d")}_{time.hour:02}00.V01.nc4'
+            url  = f'{root.format(time.year, time.month, time.day)}/{base}'
+            session = requests_retry_session()
+            resp    = session.get(url, stream=True)
+            assert resp.ok, f'Could not access url for time: {time}'
+            out     = out.replace('h5', 'nc')
+            with open(out, 'wb') as fh:
+                shutil.copyfileobj(resp.raw, fh)
+            with h5py.File(out, 'r') as ds:
+                q = ds['QV'][0, :, lat_min_ind:(lat_max_ind + 1), lon_min_ind:(lon_max_ind + 1)]
+                p = ds['PL'][0, :, lat_min_ind:(lat_max_ind + 1), lon_min_ind:(lon_max_ind + 1)]
+                t = ds['T'][0, :, lat_min_ind:(lat_max_ind + 1), lon_min_ind:(lon_max_ind + 1)]
+                h = ds['H'][0, :, lat_min_ind:(lat_max_ind + 1), lon_min_ind:(lon_max_ind + 1)]
```

Comment:
Here we are overwriting the original file.
Even though the output is saved in netcdf its not CF compliant.

I suggest we only download the file and do not touch it here.
Rather update function `_load_model_level` on https://github.com/dbekaert/RAiDER/blob/05c0a8bf3bddc423136c3362c40194548db899a6/tools/RAiDER/models/gmao.py#L134-L140 to be in an if-else statement depending on historic versus recent GMAO data.

This would be in line like the ecmwf load function where there is more work done as well than loading the basic variables: https://github.com/dbekaert/RAiDER/blob/05c0a8bf3bddc423136c3362c40194548db899a6/tools/RAiDER/models/era5.py#L97